scripts/jinja_cmake_gen.py

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env python3
+import jinja2
+import argparse
+import os
+import fnmatch
+import numpy as np
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument('filename', help="file that the sdf file should be generated from")
+    parser.add_argument('env_dir')
+    args = parser.parse_args()
+    env = jinja2.Environment(loader=jinja2.FileSystemLoader(args.env_dir))
+    template = env.get_template(os.path.relpath(args.filename, args.env_dir))
+
+    try:
+        import rospkg
+        rospack = rospkg.RosPack()
+    except ImportError:
+        pass
+        rospack = None
+
+    d = {'np': np, 'rospack': rospack, \
+         'sdf_version': "NotSet", \
+         'mavlink_tcp_port': 4560, \
+         'mavlink_udp_port': 14560, \
+         'serial_enabled': 0, \
+         'serial_device': "/dev/ttyACM0", \
+         'serial_baudrate': 921600, \
+         'enable_lockstep': 1, \
+         'model_name': "NotSet", \
+         'hil_mode': 0}
+
+    result = template.render(d)
+    filename_out = args.filename.replace('.sdf.jinja', '.sdf')
+
+    with open(filename_out, 'w') as f_out:
+        print(('{:s} -> {:s}'.format(args.filename, filename_out)))
+        f_out.write(result)
```

Comment:
Maybe I am just not understanding this correctly, but why is this needed if we have `jinja_model_gen.py`?

Reply:
This is to decouple potential errors in the build from new features in the "full options" generator. This is intentionally meant to be extremely simple and used exclusively to build all the defaults with cmake, the other is for user optional generation. We can remove this in the future if desired but for now I think it is important to keep.

Reply:
@bperseghetti Okay, but my comment on having two redundant scripts was precisely coming from having two ways of doing the same functionality and one being untested. I would prefer if the full blown script is still part of the pipeline. Otherwise this will just rot unless somebody is actively using it

Reply:
As far as I understand, `jinja_cmake_gen.py` is completely redundant functionally with `jinja_model_gen.py`. However, while `jinja_cmake_gen.py` is tested in our CI pipeline(we run SITL tests with it) `jinja_model_gen.py` is not being tested. Therefore if we are to make changes in the future to `jinja_cmake_gen.py` we will need to manually copy the changes into `jinja_model_gen.py`. Therefore I want to have a single script so that it has lower maintenance overhead. OR we need to add tests that use `jinja_model_gen.py` as a separate integration test

Reply:
Keeping separate how I originally intended, there should be no issue with doing that.

Reply:
I am still not clear on what is blocking us from using the full blown script as the single script to do everything. For example when I want to improve the script, which one should I update? do I need to keep track of both?

Reply:
You should never need to worry about keeping `jinja_gen.py` up to date as any extra changes/editions can be handled with "defaults" in the individual jinja models if they are not explicitly set. And since that is how the default models are created you also do not need to worry about circle CI for jinja_model_gen.py or jinja_world_gen.py as they are "optional". I also re-added back in my other PR the gazebo_sitl_multiple_run.sh so people wanting to use that still can.

Reply:
I would find it hard to see any issue with having both as it provides a cleaner "user experience" by not breaking any current pipelines, still gets tested in CI, and the other generators can be used or not used if desired.

Reply:
I don't think this issue has been resolved. So how can we make sure the generation scripts are being maintained?